docs/processing-data.org

```diff
@@ -0,0 +1,104 @@
+
+** Gathering and Processing Data
+ The steps for our data flow are as follows:
+ - Update our sources.yaml file -which details which audit logs to draw from and where they're located- so that it contains the most recent jobs.
+ - Download the raw logs specified in the =sources.yaml= to a local =cache/= folder.
+ - Process  the logs in our =cache= to produce new, apisnoop-ready JSON files and store them in =cache/processed=
+ - Upload these processed logs to APISnoop's GCS bucket, so they can be consumed by our webapp
+
+   We have a helper script in the root of our repo called =apisnoop.sh= which runs through each of these steps, or individual ones as specified.  To process the data on your own, it is best to run the script (detailed below).
+
+** Displaying the Data online
+   Our webapp consists of a backend server and frontend client.  The process for displaying the data is as follows:
+- Download our processed files from the GCS bucket to the local =data-gen/processed= folder within the backend server.
+- Consume all files within =data-gen/processed= to populate the backend API, slightly adjusting or combining some of the data to be easier for the frontend to use.
+- When someone visits the site, ping the backend based on whichever release has been requested and display the endpoints, tests, and metadata for that release.
+
+Our helper script, =apisnoop.sh= handles the download of the processed files.   The rest of the steps are built into the server and client.
+* Our helper script
+  When in the root of our repo, you can run =./apisnoop.sh= to get our helper script.  Running =./apisnoop --help= will displays the arguments you can do.  It is best to read that, as it will have the most up-to-date info.
+
+There are a couple caveats:
+
+- The locations we draw our audit logs are hardcoded as:
+  - gcsweb = "https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/"
+  - storage = "https://storage.googleapis.com/kubernetes-jenkins/logs/"
+- The backend is held in this repo at =web/backend= and it is hardcoded to expect the data it needs to be in =data-gen/processed=
+- The frontend displays, for each release, a link to the spyglass for the log, and a link to apisnoop's processed JSON. The link for the processed json is hardcoded to point to apisnoop's own gcs bucket.
+
+*  Developing apisnoop
+
+  The code for gathering and processing the data is held within =data-gen=.
+
+#+begin_example
+data-gen
+├── downloadArtifacts.py
+├── downloadAudits
+├── lib
+│   ├── __init__.py
+│   └── parsers.py
+├── processArtifacts.py
+├── processArtifacts.sh
+├── processAuditlog.py
+├── README.md
+├── requirements.txt
+├── sources.yaml
+└── updateSources.py
+
+1 directory, 11 files
+#+end_example
+
+Our helper script simply runs the proper .py scripts held here.  You can get a better orientation for the flow by checking out =apisnoop.sh= and seeing which files are run for which arguments.
+
+The backend and client are both held in =web=.
+
+#+RESULTS:
```

Comment:
I think you'll need to remove the #+RESULTS: line here as it's directly renders rather than put the code into a block.

https://github.com/cncf/apisnoop/blob/35b8b04a4e693373566b94f2f12bdb1c19a510ed/docs/processing-data.org#developing-apisnoop

![image](https://user-images.githubusercontent.com/31331/54161983-d6354f80-44b8-11e9-914f-8f035117ed7c.png)